wicket-core/src/main/java/org/apache/wicket/markup/head/AbstractJavaScriptReferenceHeaderItem.java

```diff
@@ -135,6 +135,11 @@ public boolean equals(Object o)
 	@Override
 	public int hashCode()
 	{
-		return Objects.hash(async, defer, charset);
+		// Not using `Objects.hash` for performance reasons
+		int result = super.hashCode();
```

Comment:
I just noticed some inconsistencies:

You added `super.hashCode` here, but not to `AbstractCssReferenceHeaderItem`. We should probably keep those two classes in sync.

And the bigger issue: The contracts for equals and hashCode now don't align. If we add `super.hashCode` (which imho we should, because the super class contains `Objects.hash(markupId)`), we should also add `markupId` to the equals contract.

@martin-g: Do you have any idea, why `AbstractJavaScriptReferenceHeaderItem` and `AbstractCssReferenceHeaderItem` do not currently user `super.equals` and `super.hashCode`. Do we intentionally ignore the `markupId` field in the super class? Your original commit for these implementations was 47aa1461d58b9a5660fc11032db952ce4df36a33.

Reply:
If we are unsure about this, I'll simply remove the `super.hashCode` from the new implementation to maintain the current behavior.

Reply:
I don't remember now but I guess those were generated by my IDE.
I agree that super should be called!